flow/templates/umich-greatlakes.sh

```diff
@@ -0,0 +1,31 @@
+{% extends "slurm.sh" %}
+{% block tasks %}
+{% set threshold = 0 if force else 0.9 %}
+{% set cpu_tasks = operations|calc_tasks('np', parallel, force) %}
+{% set gpu_tasks = operations|calc_tasks('ngpu', parallel, force) %}
+{% if gpu_tasks and 'gpu' not in partition and not force %}
+{% raise "Requesting GPUs requires a gpu partition!" %}
+{% endif %}
+{% set nn_cpu = cpu_tasks|calc_num_nodes(36) if 'gpu' not in partition else cpu_tasks|calc_num_nodes(40) %}
+{% set nn_gpu = gpu_tasks|calc_num_nodes(2) if 'gpu' in partition else 0 %}
+{% set nn = nn|default((nn_cpu, nn_gpu)|max, true) %}
+{% if partition == 'gpu' %}
+#SBATCH --nodes={{ nn|default(1, true)|check_utilization(gpu_tasks, 2, threshold, 'GPU') }}
+#SBATCH --ntasks-per-node={{ (gpu_tasks, cpu_tasks)|max }}
+#SBATCH --gres=gpu:{{ gpu_tasks }}
+{% else %}{# standard compute partition #}
+#SBATCH --partition=standard
+#SBATCH --nodes={{ nn|check_utilization(cpu_tasks, 36, threshold, 'CPU') }}
```

Comment:
From the user guide for greatakes, https://arc-ts.umich.edu/greatlakes/slurm-user-guide/, in the section 1.1, there is an example of job run with one node and one processor.  I wonder if the `check_utilization()` function is necessary? @Tobias-Dwyer @bdice

Reply:
And i didn't find section for setting `--cpus-per-task`. Is it default to be 1? Maybe i am wrong but I think it might be important to have the users interact with that since it is related to jobs with mpi...

Reply:
Good catch @yuanzhou0827. Because Great Lakes is "shared queue," we do not need to include a utilization check. We only need the utilization check when the schedulers assign full nodes to a job (in which case we want users to fill the whole node).

After discussion with @vyasr and @joaander, we decided to leave the current behavior which does not specify `--cpus-per-task`. signac-flow sets the environment variable `OMP_NUM_THREADS` for jobs with threaded parallelism (MPI-OpenMP hybrid operations). Our current solution matches the recommendations in the SDSC Comet documentation but differs from Great Lakes documentation. We're just going to go ahead with this as-is.

Reply:
Marking as resolved.